src/fs_event_wrap.cc

```diff
@@ -156,7 +160,27 @@ void FSEventWrap::OnEvent(uv_fs_event_t* handle, const char* filename,
   };
 
   if (filename != nullptr) {
-    argv[2] = OneByteString(env->isolate(), filename);
+    size_t len = strlen(filename);
+    // If encoding is BUFFER, return a Buffer.
+    // If encoding is UCS2, attempt to decode as UCS2 or report an error.
+    // Otherwise, Encode as the requested type, with UTF8 as the fallback.
+    if (wrap->encoding_ == BUFFER) {
+      argv[2] = Buffer::Copy(env, filename, len).ToLocalChecked();
+    } else if (wrap->encoding_ == UCS2) {
+      if (len % 2 != 0) {
+        argv[0] = Integer::New(env->isolate(), UV__EINVAL);
+        argv[2] = StringBytes::Encode(env->isolate(), filename, len, BINARY);
+      } else {
+        argv[2] = StringBytes::Encode(env->isolate(),
+            reinterpret_cast<const uint16_t*>(filename),
```

Comment:
@bnoordhuis Is it possible that `reinterpret_cast<uintptr_t>(filename) % sizeof(uint16_t) != 0`? If it is then we'd have to `memcpy` the filename into a `uint16_t*` before passing it along.